idaes/core/base/process_block.py

```diff
@@ -66,22 +70,50 @@ def _rule_default(b, *args):
             ..
 """
 
+def _get_pyomo_block_kwargs():
+    funcs=get_overloads_for(Block.__init__)
+    keywords = set()
+    for func in funcs:
+        keywords.update(inspect.getfullargspec(func).kwonlyargs)
+    return keywords
+
+_pyomo_block_keywords = _get_pyomo_block_kwargs()
 
 def _process_kwargs(o, kwargs):
     kwargs.setdefault("rule", _rule_default)
-    o._block_data_config_default = kwargs.pop("default", None)
     o._block_data_config_initialize = ConfigBlock(implicit=True)
     o._block_data_config_initialize.set_value(kwargs.pop("initialize", None))
     o._idx_map = kwargs.pop("idx_map", None)
+    _default = kwargs.pop("default", None)
+    if _default is not None:
+        deprecation_warning(
+            "The default argument for the ProcessBlock class is deprecated"
```

Comment:
It might be worth saying something like "Arguments can now be passed directly as keyword arguments" so users know how to fix this,